docs/rules.rst

```diff
@@ -0,0 +1,622 @@
+inventory. Each rule has to be defined as a YAML dict using some of
+the following keys:
+
+``type``
+  Type of the rule template used for creating the corresponding ferm
+  configuration, required. See `rule_templates`_ for a description of
+  the available rule templates.
+
+``chain``
+  Optional. ``iptables`` chain to which the rule is added or from which it
+  is removed. Defaults to ``INPUT``.
+
+``comment``
+  Optional. Comment which should be added to the generated rule configuration.
+
+``delete``
+  Optional. Delete rule from ferm configuration. Possible values ``True``
+  or ``False``. Defaults to ``False``.
+
+``domain``
+  Optional. ``iptables`` domain used for the firewall rule. Possible values:
+  ``ip``, ``ip6``. Defaults to `ferm__domains`_.
+
+``table``
+  Optional. ``iptables`` table to which the rule is added or from which it
+  is removed. Defaults to ``filter``.
+
+``filename``
+  Optional. Set custom filename for ferm rule definition instead of generated
+  one.
+
+``role``
+  Optional. Custom name used in the generated ferm rule definition file.
+
+``role_weight``
+  Optional. This allows to set the same ``weight`` for all rules of a
+  particular Ansible role.
+
+``rule_state``
+  Optional. Specify if rule is to be added or removed. Possible values:
+  ``present`` or ``absent``. Defaults to ``present``.
+
+``weight``
+  Optional. Helps with file sorting in rule directory.
+
+``weight_class``
+  Optional. Helps to manage order of firewall rules. The ``item.weight_class``
+  will be checked in the `ferm__weight_map`_ dictionary. If a corresponding
+  entry is found, its weight will be used for that rule, if not, the
+  ``item.weight`` specified in the rule will be used instead.
+
+``when``
+  Optional. Define condition for the rule to be enabled.
+
+.. _ferm__rules: defaults.html#envvar-ferm__rules
+.. _ferm__group_rules: defaults.html#envvar-ferm__group_rules
+.. _ferm__host_rules: defaults.html#envvar-ferm__host_rules
+.. _ferm__dependent_rules: defaults.html#envvar-ferm__dependent-rules
+.. _ferm__domains: defaults.html#envvar-ferm__domains
+.. _ferm__weight_map: defaults.html#envvar-ferm__weight_map
+
+.. _rule_templates:
+
+Depending on the choosen type, many additional variables are supported.
+Please check the individual template description below.
+
+
+Rule templates
+--------------
+
+There exist a number of predefined rule templates for generating firewall
+rules through ferm. Each rule definition is referencing the used template
+through its ``type`` key. The templates are located in the
+``templates/etc/ferm/ferm.d/`` directory.
+
+Following a list of the available rule templates which can be used to
+create custom rules:
+
+* ``accept``: Template to create rules that match interfaces, ports, remote
+  IP addresses/subnets and can accept the packets, reject, or redirect to
+  a different chain. The following template-specific YAML keys are supported:
+
+    ``accept_any``
+      Optional. ?
```

Comment:
The `item.accept_any` is a boolean which can be used to open or close ports conditionally, depending if a list of source addresses is specified. By default a rule will accept connections from any hosts if no restrictions are specified. If the `accept_any` is set to `False`, if there are no source addresses specified, rule will deny connections to specified port.

This allows to create different rules for services that are supposed to be "public", ie. open to all unless restricted, and services that are supposed to be "private", ie. closed to all and open only to restricted list of hosts.